internal/common/settings.go

```diff
@@ -132,3 +137,20 @@ func readFileSourceOfTrustSettings() (*FileTrustSourceSettings, error) {
 
 	return fileSettings, nil
 }
+
+func readSpireSourceOfTrustSettings() (*SpireTrustSourceSettings, error) {
+	if !viper.IsSet("trustsource.spire.domains") {
+		return nil, errors.New("trustsource.spire.domains is required but not found")
+	}
+
+	spireSettings := new(SpireTrustSourceSettings)
+	spireSettings.SpireEndpoints = viper.GetStringMapString("trustsource.spire.domains")
+	viper.SetDefault("trustsource.spire.certLocation", "/tmp/vault-spire-certs.json")
```

Comment:
These seem awkward, what's going on here?

Reply:
What do you mean by awkward?